src/AggregateFunctions/AggregateFunctionSumMap.cpp

```diff
@@ -298,12 +298,13 @@ class AggregateFunctionMapBase : public IAggregateFunctionDataHelper<
                     Field value = values[col_idx];
 
                     /// Compatibility with previous versions.
-                    if (value.getType() == Field::Types::Decimal32)
+                    WhichDataType value_type(values_types[col_idx]);
+                    if (value_type.isDecimal32())
                     {
                         auto source = value.safeGet<DecimalField<Decimal32>>();
                         value = DecimalField<Decimal128>(source.getValue(), source.getScale());
                     }
-                    else if (value.getType() == Field::Types::Decimal64)
+                    else if (value_type.isDecimal64())
```

Comment:
You changed only the serialization part. Will it work for data parts with `DateTime64` serialized in broken versions? Is it backward compatible?

Reply:
There shouldn't be such data parts, the broken versions threw an exception when trying to serialize DateTime64.

Other than DateTime64, this shouldn't have any effect because AFAICT there are no other cases where field type is Decimal{32,64} but DataType is not isDecimal{32,64}().